### ISSUE TITLE
HttpServerInventoryView being created multiple times on Broker startup

### ISSUE THREAD

**Issue by capistrant** [apache/druid#13001]
### Affected Version

0.22.2

### Description

Starting up a broker with HTTP typed serverview has been resulting with multiple HttpServerInventoryView objects being created. This would seem to create a bunch of duplicate data on heap + a bunch of duplicate HTTP requests to data servers

```
2022-08-30T18:02:20,058 INFO [main] org.apache.druid.client.HttpServerInventoryView - Starting HttpServerInventoryView.
2022-08-30T18:02:20,175 INFO [CuratorDruidNodeDiscoveryProvider-ListenerExecutor] org.apache.druid.server.coordination.ChangeRequestHttpSyncer - Starting ChangeRequestHttpSyncer[https://XXXXX:8283/_1661882540175].
2022-08-30T18:02:25,510 INFO [HttpServerInventoryView-4] org.apache.druid.server.coordination.ChangeRequestHttpSyncer - [https://XXXXX:8283/_1661882540175] synced successfully for the first time.
2022-08-30T18:02:20,072 INFO [main] org.apache.druid.client.HttpServerInventoryView - Started HttpServerInventoryView.
2022-08-30T18:02:58,244 INFO [main] org.apache.druid.client.HttpServerInventoryView - Starting HttpServerInventoryView.
2022-08-30T18:02:58,248 INFO [main] org.apache.druid.server.coordination.ChangeRequestHttpSyncer - Starting ChangeRequestHttpSyncer[https://XXXXX:8283/_1661882578248].
2022-08-30T18:02:58,897 INFO [HttpServerInventoryView-7] org.apache.druid.server.coordination.ChangeRequestHttpSyncer - [https://XXXXX:8283/_1661882578248] synced successfully for the first time.
2022-08-30T18:02:58,253 INFO [main] org.apache.druid.client.HttpServerInventoryView - Started HttpServerInventoryView.
```

Upon further investigation it appears this only happens when Druid SQL is enabled on the broker.

**Comment by suneet-s** [comment]
Looks like this may be addressed by #12209

**Comment by capistrant** [comment]
> Looks like this may be addressed by #12209 in 0.23

:doh: bad search skills on my part. We are upgrading next week so this is perfect. Thanks for taking the time to find the fix. Sorry for the spam issue!
